uniconnect-epfl/uniconnect
# feat: linked graph view with navigation

# What I did
I linked the graph view with the rest of the app.
I implemented some basic navigation and search functionalities.
I also extracted part of the code from ForceDirectedGraph.tsx and implemented it in another file to have better modularity over components.

# How I did it
I added the component inside the ContactGraph.tsx file and added some arguments to ForceDirectedGraph to implement navigation logic.

# How to verify it
1. Run the application
2. Go in the Explore section
3. Go in the Graph View mode
4. Switch to the plain view mode
5. Switch again to the graph view mode, the layout should not have changed given that you did not modify the drag by dragging or scrolling.
6. Try to interact with the graph
7. Typing in the search bar numbers from 0 to 10 included should "highlight" corresponding nodes
8. When clicking return on the keyboard, if at least one node is selected, you should be redirected to an external profile screen dummy screen
9. When keyboard is up and you click outside of it, the keyboard should disappear
10. When clicking on a node an animation should start and a modal should appear
11. Clicking on the node's profile picture displayed on the modal should redirect to an external profile screen dummy screen

# Demo video


https://github.com/uniconnect-epfl/uniconnect/assets/81919237/cd1f8b63-6446-45d6-aeba-3321ebf7e431

# Pre-merge checklist
The changes I have introduced:
